File: backend/Models/Despesa.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models;

public class Despesa
{
    public int Id { get; set; }
    
    [Required]
    [MaxLength(100)]
    public required string Categoria { get; set; }

    [Required]
    [MaxLength(100)]
    public required string Tipo { get; set; }

    [Required]
    [MaxLength(250)]
    public required string NomeDespesa { get; set; }

    [MaxLength(250)]
    public string? Descricao { get; set; }

    [Required]
    [Range(0, double.MaxValue, ErrorMessage = "O valor orçado não pode ser negativo.")]
    public decimal ValorUnitario { get; set; }

    [Required]
    [Range(0, double.MaxValue, ErrorMessage = "O valor realizado não pode ser negativo.")]
    public decimal Quantidade { get; set; }

    [Range(0, double.MaxValue, ErrorMessage = "O valor orçado não pode ser negativo.")]
    public decimal ValorOrcado { get; set; }

    [Range(0, double.MaxValue, ErrorMessage = "O valor realizado não pode ser negativo.")]
    public decimal ValorRealizado { get; set; }

    public int ProjetoId { get; set; }

    public Projeto? Projeto { get; set; }
}

File: android/app/src/main/java/com/example/radarfinanceiro/models/Despesa.java
package com.example.radarfinanceiro.models;

public class Despesa {

    private int id;
    private String categoria;
    private String tipo;
    private String nomeDespesa;
    private String descricao;
    private double valorUnitario;
    private double quantidade;
    private double valorOrcado;
    private double valorRealizado;
    private int projetoId;

    public int getId() {
        return id;
    }

    public String getCategoria() {
        return categoria;
    }

    public String getTipo() {
        return tipo;
    }

    public String getNomeDespesa() {
        return nomeDespesa;
    }

    public String getDescricao() {
        return descricao;
    }

    public double getValorUnitario() {
        return valorUnitario;
    }

    public double getQuantidade() {
        return quantidade;
    }

    public double getValorOrcado() {
        return valorOrcado;
    }

    public double getValorRealizado() {
        return valorRealizado;
    }

    public int getProjetoId() {
        return projetoId;
    }
}